components/adblock_rust_ffi/src/lib.rs

```diff
@@ -507,7 +510,15 @@ pub unsafe extern "C" fn convert_rules_to_content_blocking(rules: *const c_char)
     filter_set.add_filter_list(&rules, ParseOptions { rule_types: RuleTypes::NetworkOnly, ..Default::default() });
     // `unwrap` is safe here because `into_content_blocking` only panics if the `FilterSet` was not
     // created in debug mode
-    let (cb_rules, _) = filter_set.into_content_blocking().unwrap();
+    let (mut cb_rules, _) = filter_set.into_content_blocking().unwrap();
+    if cb_rules.len() > MAX_CB_LIST_SIZE {
+        // Note that the last rule is always the first-party document exception rule, which we want
+        // to keep. Otherwise, we can arbitrarily truncate rules before that to ensure that the
+        // list can actually compile.
+        cb_rules.swap(cb_rules.len() - 1, MAX_CB_LIST_SIZE - 1);
+        cb_rules.truncate(MAX_CB_LIST_SIZE);
+        *truncated = true;
+    }
```

Comment:
I think you need an `else { *truncated = false' }` here?

Reply:
I guess the intended use of the API was to initialize a new `bool` for each call, but this certainly doesn't hurt either